narwhal/primary/src/tests/synchronizer_tests.rs

```diff
@@ -11,20 +14,448 @@ use std::{
     time::Duration,
 };
 use test_utils::{make_optimal_signed_certificates, CommitteeFixture};
-use tokio::sync::watch;
+use tokio::sync::{oneshot, watch};
 use types::{error::DagError, Certificate, PreSubscribedBroadcastSender};
 
+#[tokio::test]
+async fn accept_certificates() {
```

Comment:
Could you please add some testing around the `push_certificates` method behaviour ? To ensure that is indeed doing the backoffs and edge case scenarios as expected

Reply:
Makes sense. I'm trying to mostly mimic the existing logic. Can I have the refactor and testing in another PR?